File: 3371-harshad-number/harshad-number.cpp
class Solution {
public:

    int sumOfDigits(int x){
        int d = 0;
        while(x != 0){
            d+=(x%10);
            x/=10;
        }
        return d;
    }

    int sumOfTheDigitsOfHarshadNumber(int x) {
        int d = sumOfDigits(x);
        if(x%d == 0){
            return d;
        }
        return -1;
    }
};

File: 3371-harshad-number/harshad-number.java
class Solution {
    private int sumOfTheDigits(int x){
        int sum = 0;
        while(x > 0){
            sum+=(x%10);
            x/=10;
        }
        return sum;
    }
    public int sumOfTheDigitsOfHarshadNumber(int x) {
        int d = sumOfTheDigits(x);
        if(x%d == 0){
            return d;
        }
        return -1;
    }
}